tests/test_api.py

```diff
@@ -38,132 +38,171 @@ def test_phone_registration_sends_message(client, mocker):
-    response_data = json.loads(json.dumps(response.data))
-    assert response.status_code == 400
-    assert response_data["non_field_errors"][0] == "Security code is already verified"
-
-    # Security code verification is not restricted to one time
-    settings.DJANGO_SETTINGS["PHONE_VERIFICATION"][
-        "VERIFY_SECURITY_CODE_ONLY_ONCE"
-    ] = False
-    response = client.json.post(url, data=data)
-    response_data = json.loads(json.dumps(response.data))
-    assert response.status_code == 200
-    assert response.data["message"] == "Security code is valid."
+def test_security_code_session_token_verification_api(client, backend):
+    with override_settings(PHONE_VERIFICATION=backend):
+        f.create_verification(
+            security_code=SECURITY_CODE,
+            phone_number=PHONE_NUMBER,
+            session_token=SESSION_TOKEN,
+        )
+        url = reverse("phone-verify")
+        data = {
+            "phone_number": PHONE_NUMBER,
+            "security_code": SECURITY_CODE,
+            "session_token": SESSION_TOKEN,
+        }
+        response = client.json.post(url, data=data)
+        assert response.status_code == 200
+        assert response.data["message"] == "Security code is valid."
+
+
+def test_phone_verification_with_incomplete_payload(client, backend):
+    with override_settings(PHONE_VERIFICATION=backend):
+        f.create_verification(
+            security_code=SECURITY_CODE,
+            phone_number=PHONE_NUMBER,
+            session_token=SESSION_TOKEN,
+        )
+        url = reverse("phone-verify")
+        data = {"phone_number": PHONE_NUMBER}
+        response = client.json.post(url, data=data)
+        assert response.status_code == 400
+        response_data = json.loads(json.dumps(response.data))
+        assert response_data["session_token"][0] == "This field is required."
+        assert response_data["security_code"][0] == "This field is required."
+
+        data = {"security_code": SECURITY_CODE}
+        response = client.json.post(url, data=data)
+        assert response.status_code == 400
+        response_data = json.loads(json.dumps(response.data))
+        assert response_data["phone_number"][0] == "This field is required."
+
+
+def test_phone_verification_with_incorrect_payload(client, backend):
+    with override_settings(PHONE_VERIFICATION=backend):
+        f.create_verification(
+            security_code=SECURITY_CODE,
+            phone_number=PHONE_NUMBER,
+            session_token=SESSION_TOKEN,
+        )
+        url = reverse("phone-verify")
+        # Payload with wrong session token
+        data = {
+            "phone_number": PHONE_NUMBER,
+            "security_code": SECURITY_CODE,
+            "session_token": "wrong-session-token",
+        }
+        response = client.json.post(url, data=data)
+        response_data = json.loads(json.dumps(response.data))
+        if backend.get("BACKEND").split("phone_verify.backends.")[1] in backends:
```

Comment:
We can make a small utility function for getting backend class here, and then do 

```suggestion
        if _get_backend_cls(backend) in backend_classes:
```

What are your thoughts?

Reply:
Sounds great!